pkg/controllers/oauthendpoints/oauth_endpoints_controller.go

```diff
@@ -0,0 +1,263 @@
+	}
+
+	getTLSConfigFunc := func() (*tls.Config, error) {
+		return getOAuthEndpointTLSConfig(kubeInformersForTargetNS.Core().V1().ConfigMaps().Lister())
+	}
+
+	return endpointaccessible.NewEndpointAccessibleController(
+		"OAuthServerService",
+		operatorClient,
+		endpointsListFunc, getTLSConfigFunc,
+		[]factory.Informer{
+			kubeInformersForTargetNS.Core().V1().ConfigMaps().Informer(),
+			kubeInformersForTargetNS.Core().V1().Services().Informer(),
+		},
+		recorder)
+}
+
+// NewOAuthServiceEndpointsCheckController returns a controller that checks the health of authentication service
+// endpoints.
+func NewOAuthServiceEndpointsCheckController(
+	operatorClient v1helpers.OperatorClient,
+	kubeInformersForTargetNS informers.SharedInformerFactory,
+	recorder events.Recorder,
+) factory.Controller {
+	endpointsListFn := func() ([]string, error) {
+		return listOAuthServiceEndpoints(kubeInformersForTargetNS.Core().V1().Endpoints().Lister())
+	}
+
+	getTLSConfigFunc := func() (*tls.Config, error) {
+		return getOAuthEndpointTLSConfig(kubeInformersForTargetNS.Core().V1().ConfigMaps().Lister())
+	}
+
+	return endpointaccessible.NewEndpointAccessibleController(
+		"OAuthServerServiceEndpoints",
+		operatorClient,
+		endpointsListFn, getTLSConfigFunc,
+		[]factory.Informer{
+			kubeInformersForTargetNS.Core().V1().Endpoints().Informer(),
+			kubeInformersForTargetNS.Core().V1().ConfigMaps().Informer(),
+		},
+		recorder)
+}
+
+func listOAuthServiceEndpoints(endpointsLister corev1listers.EndpointsLister) ([]string, error) {
+	var results []string
+	endpoints, err := endpointsLister.Endpoints("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get oauth service endpoints: %w", err)
+	}
+	for _, subset := range endpoints.Subsets {
+		for _, address := range subset.Addresses {
+			for _, port := range subset.Ports {
+				results = append(results, net.JoinHostPort(address.IP, strconv.Itoa(int(port.Port))))
+			}
+		}
+	}
+	if len(results) == 0 {
+		return nil, fmt.Errorf("oauth service endpoints are not ready")
+	}
+	return toHealthzURL(results), nil
+}
+
+func listOAuthServices(serviceLister corev1listers.ServiceLister) ([]string, error) {
+	var results []string
+	service, err := serviceLister.Services("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get oauth service: %w", err)
+	}
+	for _, port := range service.Spec.Ports {
+		results = append(results, net.JoinHostPort(service.Spec.ClusterIP, strconv.Itoa(int(port.Port))))
+	}
+	if len(results) == 0 {
+		return nil, fmt.Errorf("no valid oauth services found")
+	}
+	return toHealthzURL(results), nil
+}
+
+func listOAuthRoutes(routeLister routev1listers.RouteLister) ([]string, error) {
+	var results []string
+	route, err := routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get oauth route: %w", err)
```

Comment:
you sure it will be correctly interpreted by `NewEndpointAccessibleController` ?

Reply:
probably not, I'll remove the wrapper